doc/guides/spatial/guide_spatially_structured_networks.rst

```diff
@@ -1416,61 +1415,124 @@ center neuron are shown, as well as mask and connection probability.
     :start-after: #{ vislayer #}
     :end-before: #{ end #}
 
-.. TODO: adding masks need to be updated for 3.0?
+.. _ch:custom_masks:
+
+Creating custom masks
+---------------------
+
+In some cases, the built-in masks may not meet your needs, and you want to
+create a custom mask. There are two ways to do this:
+
+1. To use parameters to introduce a cut-off to the connection probability.
+2. To implement a custom mask in C++ as a module.
+
+Using parameters is the most accessible option; the entire implementation is done on
+the PyNEST level. However, the price for this flexibility is reduced connection efficiency
+compared to masks implemented in C++. Combining parameters to give the wanted behaviour may
+also be difficult if the mask specifications are complex.
+
+Implementing a custom mask in C++ gives much higher connection performance and greater freedom
+in implementation, but requires some knowledge of the C++ language. As the mask in this case
+is implemented in an extension module, which is dynamically loaded into NEST, it also requires
+some additional steps for installation.
+
+.. _sec:maskparameter:
+
+Using parameters to specify connection boundaries
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can use parameters that represent spatial distances between nodes to create a connection
+probability with mask behaviour. For this, you need to create a condition parameter that describes
+the boundary of the mask. As condition parameters evaluate to either ``0`` or ``1``, it can be
+used alone, with the ``nest.logic.conditional()`` parameter, or multiplied with another
+parameter or value, before passing it as the connection probability.
+
+As an example, we will define an elliptical connection boundary using parameters. First, we define
+variables controlling the shape of the ellipse.
+
+::
+
+   rx = 0.5   # radius in the x-direction
+   ry = 0.25  # radius in the y-direction
+
+Next, we define the connection boundary. We only want to connect to targets inside an ellipse, so
+the condition is
+
+.. math::
+
+   \frac{x^2}{r_x^2}+\frac{y^2}{r_y^2} \leq 1,
+
+where :math:`x` and :math:`y` are the distances between the source and target neuron, in x- and
+y-directions, respectively. We use this expression to define the boundary using parameters.
+
+::
+
+   x = nest.spatial.distance.x
+   y = nest.spatial.distance.y
+   lhs = x * x / rx**2 + y * y / ry**2
+   mask_param = nest.logic.conditional(lhs <= 1.0, 0.5, 0.0)
+   # Could also have defined it as
+   # mask_param = 0.5*(lhs <= 1.0)
```

Comment:
So you don't need to use `nest.logic.conditional()` at all when sending down to NEST? It is converted? This we should document elsewhere as well. Not in this PR though!

Reply:
In cases where the resulting value of `nest.logic.conditional()` is `0` if the condition evaluates to false, it is possible to just use multiplication with the condition instead. That's because the condition also is a parameter, which actually evaluates to `0` or `1`.

I suppose we should have some documentation on parameter comparators somewhere, probably in the detailed transition guide.